src/platforms/esp32/components/avm_builtins/i2c_driver.c

```diff
@@ -613,6 +622,51 @@ static NativeHandlerResult i2cdriver_consume_mailbox(Context *ctx)
     return cmd == I2CCloseCmd ? NativeTerminate : NativeContinue;
 }
 
+// The native driver API defined here is supported only for SMP builds
+#ifndef AVM_NO_SMP
+
+static Context *get_i2c_context_lock(term i2c_port, GlobalContext *global)
+{
+    if (UNLIKELY(!term_is_pid(i2c_port))) {
+        ESP_LOGW(TAG, "Given term is not a I2C port driver.");
+        return NULL;
+    }
+
+    int local_process_id = term_to_local_process_id(i2c_port);
+    Context *ctx = globalcontext_get_process_lock(global, local_process_id);
+
+    if (ctx->native_handler != i2cdriver_consume_mailbox || (ctx->platform_data == NULL)) {
+        ESP_LOGW(TAG, "Given term is not a I2C port driver.");
+        globalcontext_get_process_unlock(global, ctx);
+        return NULL;
+    }
+
+    return ctx;
+}
+
+I2CAcquireResult i2c_driver_acquire(term i2c_port, i2c_port_t *i2c_num, GlobalContext *global)
+{
+    Context *ctx = get_i2c_context_lock(i2c_port, global);
+    if (IS_NULL_PTR(ctx)) {
+        return I2CAcquireInvalidPeripheral;
+    }
+    struct I2CData *i2c_data = ctx->platform_data;
+    *i2c_num = i2c_data->i2c_num;
+    globalcontext_get_process_unlock(global, ctx);
+    return I2CAcquireOk;
+}
+
+void i2c_driver_release(term i2c_port, GlobalContext *global)
```

Comment:
As mentioned elsewhere, this function is useless with this current implementation.

Consequently, acquire can be renamed and merged with get_i2c_context_lock.